### DISCUSSION TITLE
利用Tkinter製作Wordle小遊戲

### DISCUSSION THREAD

**Discussion by awkward-willy** [1101-GS4719-C/Python-Community-GS4719-C#413] [Show and tell]
上課時曾聽老師介紹IDLE時提到它是完全由Python與其標準庫中的Tkinter撰寫而成的，正好小弟曾跟風運用Tkinter仿製了Wordle這款遊戲，所以這邊就想要來介紹一下這個*GUI工具。

GUI = Graphical User Interface 圖形使用者介面，就是用各種圖形顯示供使用者操作電腦的介面（相對於 CLI - Command-Line Interface）

[Tkinter介紹]
Tkinter 來自於 Tk 這個開放原始碼的開發工具，屬於 Python 內建的模組（不須額外pip install）。它提供許多不同的圖形介面工具，如；Label、Button、Entry......。未來想發各工具的介紹，但不確定有沒有人要看...所以先發了這個試試水溫

資料來源：[維基百科 - Tkinter](https://zh.wikipedia.org/wiki/Tkinter)

而目前我的Wordle遊戲大致完成了，應該只差在顯示已猜過字母的部分，但小弟我可能暫時不會更動了字庫的部分則是取自english-words，不過現在內含太多生難字詞，但我也實在懶得篩選了，大家也可以抓下來玩玩看，或者順手留給我個 Star?
![WordleGame](https://user-images.githubusercontent.com/58520947/157040971-ab33700a-14fb-4efc-9945-9ef9adb3fa86.png)
[字庫 - english-words](https://github.com/dwyl/english-words)
[我的WordleGame - Github](https://github.com/awkward-willy/Wordle)

**Comment by ThomasMeng0515** [comment]
之前也有玩過wordle 能做出來一樣的遊戲感覺滿厲害的 原作者好像也有公開原始碼給大家參考~

**Comment by chupochun** [comment]
也太強了，居然直接設計出一個小遊戲來，小弟佩服，看來要多多向你學習

**Comment by hapiwangy** [comment]
看到身旁的人也有在玩，就也想嘗試做出類似的東西。之前只有用過java做過GUI設計，期待自己也可以用python做出一個類似的成果來~~，感謝分享相關的資源

**Comment by pekoranoinu** [comment]
我也想學怎麼用程式設計遊戲。

**Comment by bloodtrail0511** [comment]
我想到能以HTML與CSS做出類似版面，但遊戲程式方面鑽研得還不深，總之太酷了:)

**Comment by HuanYuuuuuu** [comment]
或是可以做出一個注音版的wordle呢 這樣就比較不會有艱難字詞的問題了
(by the way 我覺得大哥你好強

**Reply by awkward-willy** [reply]
感謝你提供的意見！不過目前還沒有找到有注音的詞庫
若未來有機會會再嘗試看看的！
